### ISSUE TITLE
Queue Trigger - Consumption Plan - Goes to sleep and does not wake up

### ISSUE THREAD

**Issue by MLRichmond** [Azure/azure-functions-host#10508]
#### Check for a solution in the Azure portal
We have a simple Azure Function that is running on a Consumption ASP. The function runs fine when initially deployed or when "poked" from the Azure Portal - but after a period of time, goes to sleep and does not wake up, even if new messages arrive in the corresponding queue.

Please provide the following:

- Timestamp: 10/4/2024 2:50 PM CST
- Function App version:
- Function App name: KermitFunctions
- Function name(s) (as appropriate): QueuePackager
- Invocation ID:
- Region: Central US

#### Repro steps

Provide the steps required to reproduce the problem:

1) QueueTrigger function using Managed Identity and App Settings Connection information. 
2) Deploy function
3) Add msg to queue
4) Msg processes.
5) Wait 20-30 minutes (AppInsights log will show several "Disposing ScriptHost" messages)
6) Add msg to queue
7) Msg is never processed.

#### Expected behavior

Expected that QueueTrigger would eventually wake up and process the incoming message.

#### Actual behavior

QueueTrigger never wakes up.

#### Known workarounds

Go to Azure Portal and view Function. It will wake up and process.

**Comment by bhagyshricompany** [comment]
Hi @MLRichmond  thanks for reporting can you try with same connection string ?

**Comment by MLRichmond** [comment]
@bhagyshricompany Sorry, what do you mean by same connection string? Re-testing this morning, it's still not waking up.

**Comment by bhagyshricompany** [comment]
you are using identity for connection option? please share log details.Thanks

**Comment by MLRichmond** [comment]
Basically after 20 minutes or so of not receiving a queue message, we see this:
![image](https://github.com/user-attachments/assets/2cb4dcf4-bb4d-4e91-9ead-589f0e1c723b)
Then after that, if send in a new message, it never wakes up.

**Comment by Angelm116** [comment]
I am seeing a similar issue, but with a timer trigger function. The timer is set to trigger every 5 minutes. The function goes to sleep 20 minutes after deployment, and the timer trigger no longer fires. Similar to your case, the trigger starts firing again when I open the function in Azure Portal. When I go to Developer tools -> Kudu -> LogFiles, I see this: 

![Image](https://github.com/user-attachments/assets/e9f9d70f-b550-42d5-9a8b-299e0ce2de12)

**Comment by andrewkittredge** [comment]
we have the same problem

**Comment by bhagyshricompany** [comment]
I checked and seems working fine.![Image](https://github.com/user-attachments/assets/0d8c4dca-7313-41b5-98a5-9653a2fc50c3)

**Comment by MLRichmond** [comment]
@bhagyshricompany yes, on an initial run it works fine. The point is that when hosted in azure, after 20 minutes or so of inactivity, it gets out to sleep and never wakes again, even if additional messages come in.

Running it in a debugger isn't comparable to the failure mode. That, of course works fine.

**Comment by bhagyshricompany** [comment]
yes I got it.Will update you on this shortly.

**Comment by baumatron** [comment]
I'm seeing the same issue for the following configuration:
1. C# .NET 6 in-proc queue trigger, Microsoft.NET.Sdk.Functions 4.6.0 SDK, Microsoft.Azure.WebJobs.Extensions.Storage 5.3.3
2. Premium EP3 hosting plan.
3. Messages are processed successfully for hours or even days after initial startup
4. Eventually the host listener is stopped and the function is never triggered again despite there being queue messages present.
Invocations for the queue trigger. Last processed was at 2024-11-09 7:44:23 PM UTC-8
![Image](https://github.com/user-attachments/assets/2a3b3b82-3fd1-4eb5-beb5-de11fe1796c6)

Current messages in queue. Earliest was inserted on 2024-11-09 7:47 PM UTC-8
![Image](https://github.com/user-attachments/assets/825897f2-e4d0-470e-9d5b-84b032a8a36f)

Logs from the cloud role name show that after processing the last trigger the QueueListener was stopped, and finally it acquired the host lock blob lease despite not listening for queue messages. After the host lock was acquired, no more messages are triggered. There are no errors and the portal shows the app as running.

![Image](https://github.com/user-attachments/assets/2d877bd4-e2d5-457e-98d5-20e6bd657a63)

As an additional data point, I have other function apps with queue triggers as part of this deployment that are running on dedicated hosting plans and are not experiencing this problem.

**Comment by bhagyshricompany** [comment]
we are checking once it s done will update soon.Thanks

**Comment by tabiday** [comment]
I have the same issue, with an app on consumption plan triggered by blob storage. Please update as soon as possible

**Comment by Peter-Juhasz** [comment]
We are experiencing the same symptoms:
- Both on Flex Consumption Plan (Linux) and App Service Plan (Linux)
- Always-Ready enabled for FCP and Always On for ASP
- Both in France Central and North Europe
- Both with Storage Queue and Timer triggers
- We use Managed Identity to the Storage Queue
- .NET 9 Isolated model
- Usually after a deployment, sometimes on its own, it stops working
- Portal shows everything is fine, Functions green "Enabled", no logs indicating any error

The triggers stop working, messages are piling up in the queues, no errors or any logs, only manually restarting the app solves the issue, which is unacceptable for production workloads, and it impacts our customers.

@bhagyshricompany what is the ETA for the resolution? Or is there any workaround exists?

**Comment by surgupta-msft** [comment]
Have you all started seeing this behavior recently or that was always the case?

**Comment by SRE-Agent** [comment]
Thank you for bringing this issue to our attention.

We have reviewed the details provided and your observation regarding Azure Functions hosted on Consumption Plans going idle after inactivity (approximately 20–30 minutes) and failing to process incoming messages resonates with similar cases reported in our tracking.

### Investigation Summary
- **Identified Issue**: QueueTrigger functions fail to wake up after inactivity periods unless manually accessed via the Azure Portal.
- **Logs Confirmation**: Logs show 'Disposed ScriptHost' followed by inactivity for extended durations despite continuous queue messages.
- **Impact**:
  - Production-critical workloads are severely hampered, with manual intervention becoming necessary.
  - Function stops working after idle periods.
  - Affects users in both Consumption & Premium plans including Linux and managed identity effects.

### Current Mitigation Notes
While these failures rely on users manually interacting hosted-instance restart movements AppCheck quite causes Clar fixes awaiting longer finalized-trace (Limit QueueStreaming BackendQueue)*.
All backend-devts Possible Check-function correlating-loop missing retries-handler Endpoint specific-request Levels/streamRetry-backChecks matching ClearsControl?
(Logging sequ installed Logs+History counter-pauseEvent all aligned proper-solves).

**Comment by MLRichmond** [comment]
@surgupta-msft As far as I have been able to observe, this has always been the case.

**Comment by Dolphinsimon** [comment]
I have met the same issue; the timer trigger is set to trigger every 5 minutes. The function goes to sleep after the trigger invoked 5 times. Creating new functions in the same resource group won't solve the issue. We have found that we must create new functions in other resource groups to fix the issue.

**Comment by Dolphinsimon** [comment]
We have found that all the problematic functions are hosted in the same HomeStamp. If we are creating new functions in other resource groups (Also other HomeStamp) will fix the issue. The issue appears to be more a deployment problem rather than a code issue in function host.

**Comment by satvu** [comment]
I'm not sure all of the different comments on this thread are related to the original issue, but I think the original issue could be related to https://github.com/Azure/azure-functions-host/issues/8261 - @MLRichmond can you try the mitigation described in that thread and see if it works? The steps include:

- Explicitly defining the Connection property
- Reviewing and adding roles as needed (Data Owner, Data Sender, Data Receiver)

**Comment by dreinhardt89** [comment]
This happens periodically for us as well.

- Function stops consuming messages in queue.
- User reports issue.
- We go to visit the function in the azure portal which wakes it back up and gets it consuming messages again.
